broker/src/main/java/org/apache/rocketmq/broker/transaction/queue/TransactionalMessageBridge.java

```diff
@@ -176,7 +179,9 @@ private PullResult getMessage(String group, String topic, int queueId, long offs
             List<ByteBuffer> messageBufferList = getMessageResult.getMessageBufferList();
             for (ByteBuffer bb : messageBufferList) {
                 MessageExt msgExt = MessageDecoder.decode(bb);
-                foundList.add(msgExt);
+                if (msgExt != null) {
```

Comment:
MessageExt msgExt = MessageDecoder.decode(bb); =》 MessageExt msgExt = MessageDecoder.decode(bb, true, false);
to prevent the uncompress more that one times.